curp/src/server/raw_curp/mod.rs

```diff
@@ -551,22 +553,43 @@ impl<C: 'static + Command, RC: RoleChange + 'static> RawCurp<C, RC> {
 
         let st_r = self.st.read();
         let log_r = self.log.read();
+        let contains_candidate = self.cluster().contains(candidate_id);
+        let remove_candidate_is_not_committed =
+            log_r
+                .fallback_contexts
+                .iter()
+                .any(|(_, ctx)| match ctx.origin_entry.entry_data {
+                    EntryData::ConfChange(ref cc) => cc.changes().iter().any(|c| {
+                        matches!(c.change_type(), ConfChangeType::Remove)
+                            && c.node_id == candidate_id
+                    }),
+                    EntryData::Empty(_) | EntryData::Command(_) | EntryData::Shutdown(_) => false,
+                });
+        // extra check to shutdown removed node
+        if !contains_candidate && !remove_candidate_is_not_committed {
```

Comment:
Should this check be added to handle_vote? It it possible that a removed leader complete a pre-vote and start a vote?

Reply:
If the pre-vote is complete, there is no healthy leader in the current cluster, so it is impossible to process `MemberRemoveRequest`.